.changeset/friendly-hairs-happen.md

```diff
@@ -0,0 +1,61 @@
+---
+'@backstage/plugin-catalog-backend-module-aws': patch
+---
+
+Add a new provider `AwsS3EntityProvider` as replacement for `AwsS3DiscoveryProcessor`.
+
+In order to migrate from the **deprecated** `AwsS3DiscoveryProcessor` you need to apply
```

Comment:
The deprecation will still allow to continue using it until they get really removed eventually, but of course, it still encourages to switch.

Personally, I don't see a benefit of the existing processor over the provider.

But that is your strategy and call. I can move this out into a separate PR (or drop it for now) and create a separate documentation instead of the modification to discovery.md, too.

Let me know.

Reply:
I'd say: don't deprecate yet, but do update docs to highlight the provider as the primary path to take, demoting the processor.

The processor lets you more easily dynamically drive it through the use of locations, which arguably isn't a great use case in and of itself, but it's familiar and people use it out there so it's still "valid" even though it has some distinct drawbacks...

Reply:
alright. Then I will remove the deprecation, split the docs (recover the old and put the current one into a new page "Ingestion.md" maybe) and formulate it a bit milder.

PS: this provider also uses Locations. It is pretty similar to the processor. Which is also the case for a few others where we have both. For me as user, it is a bit hard to see why I should prefer the processor over the provider - but maybe it is just me. The argument of provider over processor would be the absence of orphans.

Reply:
Just one article (at the original filename) is fine, it'd just have an "oh and by the way you can also use a processor but it's not recommended" section with its own heading at the bottom). Look at the updates I did for https://backstage.io/docs/integrations/ldap/org for inspiration

Reply:
done